Loop/Managers/Alerts/AlertManager.swift

```diff
@@ -322,22 +371,36 @@ extension AlertManager: AlertIssuer {
 
         // Only alerts with foreground content are replayed
         if alert.foregroundContent != nil {
-            modalAlertIssuer?.issueAlert(alert)
+            modalAlertIssuer.issueAlert(alert)
         }
     }
+
+    private func issueAlertWithHandlers(_ alert: Alert, issuedDate: Date = Date()) {
+        modalAlertIssuer.issueAlert(alert)
+        userNotificationAlertIssuer.issueAlert(alert, muted: alertMuter.shouldMuteAlert(alert, issuedDate: issuedDate))
+    }
+
+    private func retractAlertWithHandlers(identifier: Alert.Identifier) {
+        modalAlertIssuer.retractAlert(identifier: identifier)
+        userNotificationAlertIssuer.retractAlert(identifier: identifier)
+    }
+
+    private func rescheduleAlertWithHandlers(_ alert: Alert, issuedDate: Date) {
+        retractAlertWithHandlers(identifier: alert.identifier)
```

Comment:
yes, but only with handlers. The alert store is not being touched in this case. Retraction pulls removed the previous alert (muted or unmuted), say from a pending iOS notification, and issuing replaces it.

Curious of what you are considering/concerned with?

Reply:
Ok, if we're not actually "retracting" (like marking it in the database as retracted) and reissuing (with another entry in the database), then that is ok from a behavior standpoint, I think. But do we want to  be calling it "retract" and "issue", if this is just updating some signalization for the same alerts? 

And the modal components don't really need to be rescheduled, do they? Keeping the same timers should work, and when the timers fire, the mute/unmute state is evaluated.

Reply:
Ok, yeah, thanks for the reminder. It'd be nice if there was a way to update the sound for a pending iOS notification, but because we have to reschedule, we'll lose the original repeat offset, which means repeats will be out-of-sync with modal presentations. Sounds find to me.